Fionn88/devops-new-crawler
# First Version of DevOps News Web Scraping Service

### Description:
This is the initial version of our DevOps News Web Scraping Service. The service is capable of scraping DevOps news articles and storing the data in a database. The functionality is divided into two modes based on the environment variable settings:

**1. First Crawl Mode:**

- When FIRST_CRAWL is set to True, the service initializes the database and performs a comprehensive crawl, capturing articles from the past six months.

**2. Daily Crawl Mode:**

- When FIRST_CRAWL is not set, the service performs a daily crawl, capturing articles from the past two days.

### Features:

- Scrapes DevOps news articles.
- Stores scraped data in a database.
- Two modes of operation (initial and daily crawl) controlled by environment variables.

### How to Test:

1. Set up the environment variable FIRST_CRAWL=True for the initial comprehensive crawl.
2. Run the service and verify that the database is populated with data from the past six months.
3. For daily operations, remove or set FIRST_CRAWL=False and verify that only articles from the past two days are added to the database.
### Notes:

Ensure to check the documentation for detailed instructions on setting up and running the service.
